### PR TITLE
Filterline rework + Restsharp lib update

### LINKED ISSUE #568: report filter: shift functionality from compiler to parser / semantic check
after parsing -> semantic check
then compiling

This needs to be shifted to parser and future semantic check code - not immediately though
```
 roles/lib/files/FWO.Report.Filter/Ast/AstNodeFilter.cs
            DateTime startOfNextMonth = startOfCurrentMonth.AddMonths(1);
            DateTime startOfPrevMonth = startOfCurrentMonth.AddMonths(-1);

            switch (timeRange)
```
and this as well:
```
roles/lib/files/FWO.Report.Filter/DynGraphqlQuery.cs
@@ -46,30 +47,74 @@ public static DynGraphqlQuery Generate(AstNode ast)
            // if any filter is set, optionally leave out all header texts

            string paramString = string.Join(" ", query.QueryParameters.ToArray());
            switch (query.ReportType)
```

### LINKED ISSUE #1210: report - filter line - only show enabled rules
provide filtering by enabled/disabled

### LINKED ISSUE #1220: UI: exception handling - add graphql query variables readable
also output variable content in case of exeption

https://github.com/CactuseSecurity/firewall-orchestrator/blob/develop/roles/lib/files/FWO.Api.Client/APIConnection.cs#L107

### LINKED ISSUE #1294: report filter line - add exact match operators
e.g. svc=all vs. svc ==all (first should match "install", second should not)

### LINKED ISSUE #1411: RestSharp Update to v107 (was: warnings and errors during dotnet build process)
- [x] warning about missing  '--self-contained' or '--no-self-contained' option
- [x] missing project reference in .sln file for autodiscovery
- [x] error after upgrading restsharp to v107:  `/usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(29,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]`

full build log:

```
    Restored /usr/local/fworch/ui/files/FWO.UI/FWO.Ui.csproj (in 4.88 sec).
    /usr/share/dotnet/sdk/6.0.101/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(1110,5): warning NETSDK1179: One of '--self-contained' or '--no-self-contained' options are required when '--runtime' is used. [/usr/local/fworch/ui/files/FWO.UI/FWO.Ui.csproj]
    /usr/share/dotnet/sdk/6.0.101/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(1110,5): warning NETSDK1179: One of '--self-contained' or '--no-self-contained' options are required when '--runtime' is used. [/usr/local/fworch/lib/files/FWO.Config.Api/FWO.Config.Api.csproj]
    /usr/share/dotnet/sdk/6.0.101/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(1110,5): warning NETSDK1179: One of '--self-contained' or '--no-self-contained' options are required when '--runtime' is used. [/usr/local/fworch/lib/files/FWO.Config.File/FWO.Config.File.csproj]
    /usr/share/dotnet/sdk/6.0.101/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(1110,5): warning NETSDK1179: One of '--self-contained' or '--no-self-contained' options are required when '--runtime' is used. [/usr/local/fworch/lib/files/FWO.Report.Filter/FWO.Report.Filter.csproj]
    /usr/share/dotnet/sdk/6.0.101/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(1110,5): warning NETSDK1179: One of '--self-contained' or '--no-self-contained' options are required when '--runtime' is used. [/usr/local/fworch/lib/files/FWO.Logging/FWO.Logging.csproj]
    /usr/share/dotnet/sdk/6.0.101/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(1110,5): warning NETSDK1179: One of '--self-contained' or '--no-self-contained' options are required when '--runtime' is used. [/usr/local/fworch/lib/files/FWO.Api.Client/FWO.Api.Client.csproj]
    /usr/share/dotnet/sdk/6.0.101/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(1110,5): warning NETSDK1179: One of '--self-contained' or '--no-self-contained' options are required when '--runtime' is used. [/usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/FWO.DeviceAutoDiscovery.csproj]
      FWO.Logging -> /usr/local/fworch/lib/files/FWO.Logging/bin/Release/net6.0/FWO.Logging.dll
      FWO.Config.File -> /usr/local/fworch/lib/files/FWO.Config.File/bin/Release/net6.0/FWO.Config.File.dll
    /usr/share/dotnet/sdk/6.0.101/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(1110,5): warning NETSDK1179: One of '--self-contained' or '--no-self-contained' options are required when '--runtime' is used. [/usr/local/fworch/lib/files/FWO.Middleware/FWO.Middleware.csproj]
      FWO.Middleware -> /usr/local/fworch/lib/files/FWO.Middleware/bin/Release/net6.0/FWO.Middleware.dll
    /usr/share/dotnet/sdk/6.0.101/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(1110,5): warning NETSDK1179: One of '--self-contained' or '--no-self-contained' options are required when '--runtime' is used. [/usr/local/fworch/lib/files/FWO.Report/FWO.Report.csproj]
    /usr/share/dotnet/sdk/6.0.101/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(1110,5): warning NETSDK1179: One of '--self-contained' or '--no-self-contained' options are required when '--runtime' is used. [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Api.Client/ApiSubscription.cs(31,21): warning CS0162: Unreachable code detected [/usr/local/fworch/lib/files/FWO.Api.Client/FWO.Api.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Api.Client/APIConnection.cs(125,25): warning CS0162: Unreachable code detected [/usr/local/fworch/lib/files/FWO.Api.Client/FWO.Api.Client.csproj]
      FWO.Api.Client -> /usr/local/fworch/lib/files/FWO.Api.Client/bin/Release/net6.0/FWO.Api.Client.dll
      FWO.Report.Filter -> /usr/local/fworch/lib/files/FWO.Report.Filter/bin/Release/net6.0/FWO.Report.Filter.dll
      FWO.Config.Api -> /usr/local/fworch/lib/files/FWO.Config.Api/bin/Release/net6.0/FWO.Config.Api.dll
    /usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(29,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(36,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(43,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(50,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(57,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(64,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(71,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(78,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(84,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(91,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(97,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(104,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(111,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(118,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(125,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(132,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(139,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(146,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(153,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(160,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(167,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(174,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(181,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(188,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(195,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(202,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(209,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.Middleware.Client/MiddlewareClient.cs(216,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.Middleware.Client/FWO.Middleware.Client.csproj]
    /usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/FortiManagerAPI.cs(29,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/FWO.DeviceAutoDiscovery.csproj]
    /usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/FortiManagerAPI.cs(48,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/FWO.DeviceAutoDiscovery.csproj]
    /usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/FortiManagerAPI.cs(64,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/FWO.DeviceAutoDiscovery.csproj]
    /usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/FortiManagerAPI.cs(82,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/FWO.DeviceAutoDiscovery.csproj]
    /usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/FortiManagerAPI.cs(100,27): error CS0246: The type or namespace name 'IRestResponse<>' could not be found (are you missing a using directive or an assembly reference?) [/usr/local/fworch/lib/files/FWO.DeviceAutoDiscovery/FWO.DeviceAutoDiscovery.csproj]
      FWO.Report -> /usr/local/fworch/lib/files/FWO.Report/bin/Release/net6.0/FWO.Report.dll
  stdout_lines: <omitted>

PLAY RECAP *********************************************************************
fworch-srv                 : ok=202  changed=134  unreachable=0    failed=1    skipped=115  rescued=0    ignored=0   

Error: Process completed with exit code 2.
```

### PR DESCRIPTION
**Pull request by NilsPur**
closes #1411 (Restsharp update)
closes #1294 (Exact equals operator)
closes #1220 (GraphQl variable readable)
closes #1210 (Disabled filter)
closes #568 (Semantic check)

### CODE REVIEW

**Review by tpurschke — CHANGES_REQUESTED** [review]
issues:
- PDF report generation causes error "Export Report - Unclassified error: : One or more errors occurred. (Unable to load native library. The platform may be missing native dependencies (libjpeg62, etc). Or the current platform is not supported.)"
- report on freshly imported rules does not work (empty ruleset) - could this be related to change in roles/lib/files/FWO.Config.Api/ConfigDbAccess.cs line 25?

**Comment by NilsPur** [comment]
Just looked into the timing problem. Fixed the issue on my sight. Does it work now @tpurschke

**Comment by tpurschke** [comment]
could not reproduce yesterday's reporting issue with empty ruleset (neither pre-fix nor post-fix).
but feel more confident with time fixed version.
Now it's only the PDF issue. can you activate the preview version?

File: roles/lib/files/FWO.Report/FWO.Report.csproj [hunk 1]
```diff
@@ -7,7 +7,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Haukcode.WkHtmlToPdfDotNet" Version="1.5.59" />
+    <PackageReference Include="Haukcode.WkHtmlToPdfDotNet" Version="1.5.68" />
```
Comment (tpurschke) [inline roles/lib/files/FWO.Report/FWO.Report.csproj:10]: pls revert this

**Review by tpurschke — CHANGES_REQUESTED** [review]
see inline comments

File: roles/lib/files/FWO.Report/FWO.Report.csproj (hunk 1 above)
Comment (NilsPur) [reply roles/lib/files/FWO.Report/FWO.Report.csproj]: Done

**Pull request merged**